pkg/arvo/app/graph-store.hoon

```diff
@@ -311,6 +315,37 @@
         (add-node-list resource graph mark (sort-nodes nodes))
       ==
       ::
+      ++  check-for-duplicates
+        |=  [=resource:store nodes=(set index:store)]
+        ^-  ?
+        =|  has-duplicates=_|
+        =/  node-list  ~(tap in nodes)
+        |-
+        ?~  node-list
+          has-duplicates
+        =?  has-duplicates  (has-node resource i.node-list)  %.y
+        $(node-list t.node-list)
+      ::
+      ++  has-node
+        |=  [=resource:store =index:store]
+        ^-  ?
+        =/  parent-graph=(unit marked-graph:store)
+          (~(get by graphs) resource)
+        ?~  parent-graph  %.n
```

Comment:
Why don't we move this up into the start of `+check-for-duplicates`, and pass it in as an argument here? No need to `(~(get by graphs) resource)` on every node check for the same graph, right?

Reply:
Good point.